gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopSyncableOutputStream.java

```diff
@@ -245,9 +245,7 @@ public void sync() throws IOException {
    */
   @Override
   public void hflush() throws IOException {
-    logger.atWarning().log(
-        "hflush() is a no-op; readers will *not* yet see flushed data for %s", finalGcsPath);
-    throwIfNotOpen();
+    hsync();
```

Comment:
We may need to put this behind the flag, especially if we want to back-port it to older GCS connector versions. We can enable it by default in `master` branch though.